apis/database/v1alpha1/types.go

```diff
@@ -98,3 +98,36 @@ type PostgreSQLInstanceList struct {
 	metav1.ListMeta `json:"metadata,omitempty"`
 	Items           []PostgreSQLInstance `json:"items"`
 }
+
+// NoSQLDBInstanceSpec specifies the desired state of a NoSQLDBInstance.
+type NoSQLDBInstanceSpec struct {
+	runtimev1alpha1.ResourceClaimSpec `json:",inline"`
+}
+
+// +kubebuilder:object:root=true
+
+// A NoSQLDBInstance is a portable resource claim that may be satisfied by binding
+// to a NoSQLDB managed resource such as an AWS DynamoDB or an Azure CosmosDB instance.
+// +kubebuilder:printcolumn:name="STATUS",type="string",JSONPath=".status.bindingPhase"
+// +kubebuilder:printcolumn:name="CLASS-KIND",type="string",JSONPath=".spec.classRef.kind"
+// +kubebuilder:printcolumn:name="CLASS-NAME",type="string",JSONPath=".spec.classRef.name"
+// +kubebuilder:printcolumn:name="RESOURCE-KIND",type="string",JSONPath=".spec.resourceRef.kind"
+// +kubebuilder:printcolumn:name="RESOURCE-NAME",type="string",JSONPath=".spec.resourceRef.name"
+// +kubebuilder:printcolumn:name="AGE",type="date",JSONPath=".metadata.creationTimestamp"
+// +kubebuilder:subresource:status
+type NoSQLDBInstance struct {
```

Comment:
nitpick: do you think it makes sense to call it `NoSQLInstance` to match `MySQLInstance` and `PostgreSQLInstance`? I think the DB part can be implied by it being in the `database` group

Reply:
Agree 100%